src/contrib/cluster/Akka.DistributedData/Replicator.cs

```diff
@@ -289,19 +289,23 @@ internal sealed class Replicator : UntypedActor, IWithUnboundedStash
         /// <summary>
         /// Cluster nodes, doesn't contain selfAddress.
         /// </summary>
-        private ImmutableHashSet<Address> _nodes = ImmutableHashSet<Address>.Empty;
+        private ImmutableSortedSet<Address> _nodes = ImmutableSortedSet<Address>.Empty;
+
+        // cluster members sorted by age, oldest first,, doesn't contain selfAddress, doesn't contain joining and weaklyUp
+        // only used when prefer-oldest is enabled
+        private ImmutableSortedSet<Member> _membersByAge = ImmutableSortedSet<Member>.Empty.WithComparer(Member.AgeOrdering);
 
-        private ImmutableHashSet<Address> AllNodes => _nodes.Union(_weaklyUpNodes);
+        private ImmutableSortedSet<Address> AllNodes => _nodes.Union(_weaklyUpNodes);
 
         /// <summary>
         /// Cluster weaklyUp nodes, doesn't contain joining and not selfAddress
         /// </summary>
-        private ImmutableHashSet<Address> _weaklyUpNodes = ImmutableHashSet<Address>.Empty;
+        private ImmutableSortedSet<Address> _weaklyUpNodes = ImmutableSortedSet<Address>.Empty;
```

Comment:
Moving to sorted set - part of the deterministic sort order mentioned earlier. Got it.